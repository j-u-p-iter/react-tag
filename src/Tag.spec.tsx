import * as React from "react";
import { cleanup, fireEvent, render } from "react-testing-library";

import { Tag, TagProps } from ".";

describe("Tag", () => {
  let getTagEl: any;

  beforeAll(() => {
    getTagEl = (props: TagProps) => (
      <Tag className="some-class-name" {...props} />
    );
  });

  afterEach(cleanup);

  it("renders correctly by default", () => {
    const { container } = render(getTagEl());

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with all props", () => {
    const { container } = render(
      getTagEl({
        isClearable: true,
        onClick: () => {}
      })
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("handles click correctly", () => {
    const { getByTestId, container } = render(
      getTagEl({
        isClearable: true
      })
    );

    expect(container.firstChild).toBeDefined();

    fireEvent.click(getByTestId("icon"));

    expect(container.firstChild).toBe(null);
  });

  it("calls onClick correctly", () => {
    const onClick = jest.fn();

    const { getByTestId } = render(
      getTagEl({
        isClearable: true,
        onClick
      })
    );

    fireEvent.click(getByTestId("icon"));

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick.mock.calls[0][0].currentTarget).toBeDefined();
  });
});
