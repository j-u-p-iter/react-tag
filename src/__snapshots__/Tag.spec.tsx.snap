// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly by default 1`] = `
<span
  class="some-class-name"
>
  <span>
    some text
  </span>
</span>
`;

exports[`Tag renders correctly with all props 1`] = `
<span
  class="some-class-name"
>
  <span>
    some text
  </span>
  <i
    data-testid="icon"
  >
    x
  </i>
</span>
`;
